src/main/kotlin/lgp/core/evolution/population/Models.kt

```diff
@@ -3,13 +3,44 @@ package lgp.core.evolution.population
 import lgp.core.environment.CoreModuleType
 import lgp.core.environment.Environment
 import lgp.core.environment.dataset.Dataset
+import lgp.core.evolution.ExportableResult
 import lgp.core.evolution.fitness.Evaluation
 import lgp.core.evolution.fitness.FitnessEvaluator
 import lgp.core.modules.ModuleInformation
 import java.util.Random
 import kotlin.concurrent.thread
 import kotlin.streams.toList
 
+/**
+ * An [ExportableResult] implementation that represents evolution statistics for a particular run.
+ *
+ * Because evolution statistics are collected on a per-generation basis, this type of exportable result
+ * collects data on each generation for each run.
+ *
+ * @param run The run this result relates to.
+ * @param statistics Evolution statistics for a particular generation.
+ */
+class RunBasedExportableResult<T>(
+    val run: Int,
+    private val statistics: EvolutionStatistics
+) : ExportableResult<T> {
+
+    /**
+     * Exports this result as a mapping of statistic names to statistic values.
+     */
+    override fun export(): List<Pair<String, String>> {
+        val out = mutableListOf(
+                Pair("run", this.run.toString())
+        )
+
+        out.addAll(statistics.data.map { (name, value) ->
```

Comment:
There might be a more elegant way to map between these types (as they are pretty similar).